src/golang/lib/execution_environment/execution_environment.go

```diff
@@ -202,13 +213,16 @@ func SyncEnvsWithDB(
 
 		// Env is missing
 		if err == database.ErrNoRows {
-			env.CreateDBRecord(ctx, envWriter, db)
+			err = env.CreateDBRecord(ctx, envWriter, db)
```

Comment:
@likawind I was thinking starting a transaction within this API to ensure atomicity. I'll make some minor tweak in my branch and let you review.

Reply:
I generally agree with the transaction idea. What I'm thinking is actually to let the caller pass in a transaction object and control the commit and rollback on that level. In this way the caller can control the atomicity on broader level (together with integration / workflow object creation for example).